### PR TITLE
new script for make data count log processing as well as updated docu…

### PR DESCRIPTION
**Pull request by stevenwinship**
new script for make data count log processing as well as updated documentation for counter-processor

PR adds automation for processing log files for metrics capture and send to DataCite (Make Data Count)

Closes # https://github.com/IQSS/dataverse.harvard.edu/issues/3

**Special notes for your reviewer**: 

**Suggestions on how to test this**: Test files have been added to ./tests/data/app-1 and app-2. To properly test, new logs should be generated, tar-ed, and added to these directories so the processed logs will match datasets in your test db. counter-processor needs to be installed. 

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**: No

**Is there a release notes update needed for this change?**: Added

**Additional documentation**: Multiple documents regarding counter-processor have been updated

### CODE REVIEW

File: doc/sphinx-guides/source/installation/prerequisites.rst [hunk 1]
```diff
@@ -457,22 +457,23 @@ The process required to sign up, download the database, and to configure automat
 
 As root, change to the Counter Processor directory you just created, download the GeoLite2-Country tarball from MaxMind, untar it, and copy the geoip database into place::
 
-        <download or move the GeoLite2-Country.tar.gz to the /usr/local/counter-processor-0.1.04 directory>
+        <download or move the GeoLite2-Country.tar.gz to the /usr/local/counter-processor-1.05 directory>
         tar xvfz GeoLite2-Country.tar.gz
         cp GeoLite2-Country_*/GeoLite2-Country.mmdb maxmind_geoip
+        Note: GeoLite2-Country_20191217 is already included in the installation. You can skip the download and untar if you use this version. Simply 'cp maxmind_geoip/GeoLite2-Country_20191217/GeoLite2-Country.mmdb maxmind_geoip
```
[the remark this answers is not in the document]
Comment (stevenwinship) [reply doc/sphinx-guides/source/installation/prerequisites.rst:463]: I'll remove it. I will need to remove it from the counter-processor project as well

File: scripts/makedatacount/process_mdc_logs.sh [hunk 2]
```diff
@@ -0,0 +1,167 @@
+#! /bin/bash
```
[the remark this answers is not in the document]
Comment (stevenwinship) [reply scripts/makedatacount/process_mdc_logs.sh:1]: It's designed to process archived files. We (Harvard) will be processing only files that have been archived so we won't be running the daily script. Since I don't see any mention of the script that archives the logs I didn't know if this should be documented for anyone but us.  The daily script doesn't give us the monitoring and processing control that this new script can give.
@landreev - do you have an opinion of how the two scripts should be documented, if at all? The other script isn't checked into the repo, which I think it should be.

[the remark this answers is not in the document]
Comment (landreev) [reply scripts/makedatacount/process_mdc_logs.sh:1]: @stevenwinship 
Going through my GitHub mentions, just saw the above now, sorry. 

I don't have a super strong opinion. But generally, things that are super Harvard/IQSS-specific should not be checked in the main repo. For our own housekeeping purposes we can check things in under the https://github.com/IQSS/dataverse.harvard.edu repo. Or simply document it in the "Custom Scripts and Basic Tasks on the Production Servers" Google doc. 

As @qqmyers mentioned, in some cases it can be useful to check in scripts like this and document them, as potential examples that other instances may find useful. I'm not sure about this specific script though - it may just be too exotic for that purpose. For example, having to do this on two different server nodes is a complication that may be unique to us.

[the remark this answers is not in the document]
Comment (stevenwinship) [reply scripts/makedatacount/process_mdc_logs.sh:1]: Will be moving the script to the Harvard dataverse repo

[the remark this answers is not in the document]
Comment (qqmyers) [reply scripts/makedatacount/process_mdc_logs.sh]: @stevenwinship - does that mean this PR should be closed? (It's 'Ready for Review' right now.)

[the remark this answers is not in the document]
Comment (stevenwinship) [reply scripts/makedatacount/process_mdc_logs.sh]: No. The script was moved but this fix updates the counter-processor and the nightly script and docs. This is needed to include the counter-processor bug fix that prompted the new version that dataverse will use.

File: tests/data/app-1/counter_2024-02.tar
Comment (qqmyers) [inline tests/data/app-1/counter_2024-02.tar:1]: Are these files still relevant to the PR? I don't see any test and they presumably test the script that is now in the other repo.

Comment (stevenwinship) [reply tests/data/app-1/counter_2024-02.tar:1]: Correct. Not applicable to this PR since the script move. Removing

**Review by qqmyers — APPROVED** [review]
Looks good. This PR is now ~docs-only - it updates the docs and some code comments to reference the latest version of counter-processor and the new gdcc repo keeping the updated version that was fixed as part of the overall work on the issue.

**Comment by landreev** [comment]
@stevenwinship Just double-checking, what is the scope of the QA for this PR - do I need to be testing the actual scripts (that are checked-in in the local repo)? Or is just to review the documentation in this PR?

**Comment by stevenwinship** [comment]
> @stevenwinship Just double-checking, what is the scope of the QA for this PR - do I need to be testing the actual scripts (that are checked-in in the local repo)? Or is just to review the documentation in this PR?

I would think since counter_processor has moved to gdcc and has a new release version the install instruction are important. The scripts that are here only change the version number. The other scripts have already been merged into dataverse.harvard
